CONTRIBUTING.md

```diff
@@ -2,3 +2,78 @@ Contributing to KubOS
 ---
 
 Before a pull request is merged, you must [sign our Contributor's License Agreement](https://www.clahub.com/agreements/openkosmosorg/KubOS)
+
+## Instructions:
+  1. Fork and clone the repo
+  2. Create a virtualenv for this project
+  3. Install dependent packages with pip install -e .
+  4. Install test dependent packages with pip install -r requirements-test.txt
+  5. Run tests
+  6. Submitting code
+
+### Fork and clone the repo
+
+Start by clicking the Fork button on our Github repo. Then, open a command line and type
+the following.
+
+```
+cd ~/Projects
+git clone git@github.com:{YOUR_USERNAME}/kubos-cli.git
+```
+
+### Creating a virtual environment
+
+Kubos uses Vagrant to create environments that also has the added benifit of a virtual dev environment. We will be using Vagrant in place of installing a Python Virtual Environment.
+
+#### Set up KubOS Sdk
+Follow these instructions for setting up [Vagrant for Kubos](http://docs.kubos.co/0.2.2/md_docs_sdk-installing.html)
+
+#### Linking your volume
+To develop locally on your machine and have the changes take affect, link your cloned kubos-cli folder to Vagrant.
+
+Change this line:
+```
+config.vm.synced_folder "../data", "/vagrant_data"
+```
+To this
+```
+config.vm.synced_folder "~/Projects/kubos-cli, "/home/vagrant/kubos-cli"
+```
+
```

Comment:
If a users vagrant vm is already running and they mount a new directory via the Vagrantfile they will have to run `vagrant reload` in order for the changes to their Vagrantfile to take effect in their vm. A note about that here would be helpful.